### ISSUE TITLE
POP3 SSL failure

### ISSUE THREAD

**Issue by Bluebeerd** [jstedfast/MailKit#972]
We have been trying to connect via TLS to a POP3 account and keep getting a timeout error.
We have tried several email accounts and they work perfectly when using SecureSocketOptions.None, but fail with the StartTls option.

We were originally trying to connect to a CPanel POP3 account, but have switched to an Outlook.com account to see if there was any success using a different server.

    using (var client = new Pop3Client())
    {
            client.SslProtocols = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
            client.Connect("outlook.office365.com", 995, MailKit.Security.SecureSocketOptions.StartTls);
    	// Errors on above Connect statement after 2 minutes
           client.Authenticate("discoveryssltest@outlook.com", "PASSWORD");

           for (int i = 0; i < client.Count; i++)
           {
       	          var message = client.GetMessage(i);
           }
           client.Disconnect(true);
    }

The error we are getting is:

* client.Connect("outlook.office365.com", 995, MailKit.Security.SecureSocketOptions.StartTls);
**Error**
System.IO.IOException: 'A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond'

The settings we are using are below, so can anyone else manage to make a connection ?

![Outlook_settings](https://user-images.githubusercontent.com/19287848/72523981-d92b6a00-3858-11ea-8662-1cdd35d8599c.png)

**Comment by jstedfast** [comment]
The first thing to understand is that STARTTLS is not TLS. In fact, STARTTLS has nothing to do with SSL vs TLS. STARTTLS is a command/extension implemented by many mail servers that allow a client to connect in plain-text mode and then switch into SSL/TLS mode after reading the greeting and getting the server capabilities.

The second thing to know is that no mail servers support SSL encryption anymore. They all only support TLS if they support transport encryption at all. SSL is completely obsolete across the entire web. Even early versions of TLS are obsolete at this point and TLS superseded SSL.

And finally, to fix your problem... Let me start by defining the ports for the various mail protocols:

|Protocol|Standard Port|SSL/TLS Port|
|:------:|:-----------:|:------:|
| SMTP   | 25 or 587   | 465    |
| POP3   | 110         | 995    |
| IMAP   | 143         | 993    |

The SSL/TLS port *requires* `SecureSocketOptions.SslOnConnect` which simply means that MailKit should expect the initial greeting to be encrypted via SSL/TLS.

The `StartTLS` options are for use with the *standard* ports (e.g. 110 for POP3) and tells MailKit to enable SSL/TLS encryption via the `STARTTLS` command after reading the greeting and capabilities of the server.

TL;DR:

To connect to port 995, you need to use `SecureSocketOptions.SslOnConnect`.

**Comment by Bluebeerd** [comment]
Ok, we changed the StartTLS line to
client.Connect("outlook.office365.com", 995, MailKit.Security.SecureSocketOptions.SslOnConnect);

and we still get an error

MailKit.Security.SslHandshakeException: 'An error occurred while attempting to establish an SSL or TLS connection.

One possibility is that you are trying to connect to a port which does not support SSL/TLS.

**Comment by jstedfast** [comment]
The problem is that your local SSL certificate database doesn't have the root CA for the server's certificate chain.

Try this:

```csharp
client.ServerCertificateValidationCallback = (s,c,h,e) => true;
client.Connect("outlook.office365.com", 995, MailKit.Security.SecureSocketOptions.SslOnConnect);
```
